sdk/resource/env.go

```diff
@@ -0,0 +1,68 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resource
+
+import (
+	"context"
+	"errors"
+	"os"
+	"strings"
+
+	"go.opentelemetry.io/otel/api/kv"
+)
+
+// envVar is the environment variable name OpenTelemetry Resource information can be assigned to.
+const envVar = "OTEL_RESOURCE_LABELS"
+
+var (
+	//ErrMissingValue is returned when a resource value is missing.
+	ErrMissingValue = errors.New("missing value")
```

Comment:
As this is specific to this Detector, we can probably keep it unexported. Additionally, it should wrap one of the main errors.

```suggestion
	//errMissingValue is returned when a resource value is missing.
	errMissingValue = fmt.Errorf("%w: missing value", ErrMissingResource)
```

Or, it might be a better idea to wrap `ErrPartialResource` and in `constructOTResources` omit the invalid value, but return a partial resource.

```suggestion
	//errMissingValue is returned when a resource value is missing.
	errMissingValue = fmt.Errorf("%w: missing value", ErrPartialResource)
```

Reply:
Yeah, I think ErrPartialResource is better semantically.